packages/teams-js/src/public/pages.ts

```diff
@@ -69,6 +70,24 @@ export namespace pages {
     });
   }
 
+  export function navigateToApp(params: NavigateToAppParams): Promise<void> {
+    return new Promise<void>(resolve => {
+      ensureInitialized(
+        FrameContexts.content,
+        FrameContexts.sidePanel,
+        FrameContexts.settings,
+        FrameContexts.task,
+        FrameContexts.stage,
+        FrameContexts.meetingStage,
+      );
+      if (runtime.isLegacyTeams) {
```

Comment:
I think we should just check for `runtime.supports.navigation` or whatever capability the new API falls into. Also, they way you have it right now it's possible the hub does not support `deepLink` capability but we would send that message anyway.